### PR TITLE
Fix count_nonzero and argmin/argmax

### LINKED ISSUE #585: cupy.count_nonzero does not work correctly
count_nonzero(x) should return the count of non-zero elements in x. The current implementation is not correct.

```python
>>> a = cupy.eye(4)
>>> cupy.count_nonzero(a)
1
```

Note that this bug is overlooked because the test also has a bug. `testing.shaped_random` with dtype=bool almost always returns an array filled by False. It seems a bug of shaped_random(dtype=bool), since it should return a random boolean array. We should fix it, too.

### PR DESCRIPTION
**Pull request by beam2d**
Fix #585.

### CODE REVIEW

**Comment by beam2d** [comment]
After fixing shaped_random, I found that cupy.argmin and cupy.argmax also behave different from those in numpy, when there are multiple maximum values in the target axis. I also fixed it in this PR.

File: tests/cupy_tests/sorting_tests/test_count.py [hunk 1]
```diff
@@ -9,13 +12,12 @@ class TestCount(unittest.TestCase):
...
-        m = testing.shaped_random((2, 3), xp, xp.bool_)
-        a = testing.shaped_random((2, 3), xp, dtype) * m
-        c = xp.count_nonzero(a)
-        self.assertIsInstance(c, int)
-        return c
+    def test_count_nonzero(self, dtype):
+        m = testing.shaped_random((2, 3), numpy, bool)
+        a = testing.shaped_random((2, 3), numpy, dtype) * m
+        c = numpy.count_nonzero(a)
+        d = cupy.count_nonzero(cupy.array(a))
```
Comment (unnonouno) [inline tests/cupy_tests/sorting_tests/test_count.py]: Do we need to check types of `c` and `d`?

Comment (beam2d) [inline tests/cupy_tests/sorting_tests/test_count.py]: I added a type test.

**Comment by beam2d** [comment]
I reverted the test code.

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**